### PR TITLE
[core] Fix for std::vector operator <<

### PR DESCRIPTION
**Pull request by maceligueta**
A namespace was missing.
Bug noted by @philbucher in #849.

### CODE REVIEW

File: kratos/utilities/stl_io.h
Comment (maceligueta) [inline kratos/utilities/stl_io.h]: @roigcarlo : shouldn't we replace these `std::cout` by `os` (the `std::ostream`) ??

Comment (maceligueta) [reply kratos/utilities/stl_io.h]: already talked by phone

**Comment by maceligueta** [comment]
I just pushed a new version which will work with the logger (now it is using the specific `std::cout`).
I also removed the comma at the end.

**Comment by philbucher** [comment]
Thanks @maceligueta !
I am trying it now

File: kratos/utilities/stl_io.h
Comment (philbucher) [inline kratos/utilities/stl_io.h]: if you use `(std::size_t i=1; ...)` then you don't need the casting above

File: kratos/utilities/stl_io.h
Comment (philbucher) [inline kratos/utilities/stl_io.h]: I personally don't like abusing ints as booleans, but your call
alternative would be: `if(vector_size>0) os << data[0];`

**Review by philbucher — DISMISSED** [review]
I tried it, it works
thanks @maceligueta

**Comment by maceligueta** [comment]
I tried after my last commits and it still works. Needs re-approval.

**Review by philbucher — DISMISSED** [review]
Thx!

**Comment by philbucher** [comment]
please wait with merging I am adding a test

File: kratos/tests/utilities/test_stl_io.cpp
Comment (philbucher) [inline kratos/tests/utilities/test_stl_io.cpp]: for some reason this is not working in this file, it is again not recognizing the overloaded operator
It works if I use it in other files, but not here for some reason

@maceligueta @roigcarlo any ideas?

Comment (maceligueta) [reply kratos/tests/utilities/test_stl_io.cpp]: Apparently, if you put 
`using Kratos::operator<<;`
(after `namespace Testing {`  ) it compiles. 
It looks like nesting namespaces creates the problem. I wrote a dummy function in the Kratos namespace (same file) but outside Testing namespace and it worked as well.

Comment (philbucher) [reply kratos/tests/utilities/test_stl_io.cpp]: thx @maceligueta now it works

**Review by philbucher — DISMISSED** [review]

**Review by philbucher — APPROVED** [review]

**Pull request merged**
